pkg/services/vmoperator/vmopmachine.go

```diff
@@ -63,33 +66,34 @@ func (v *VmopMachineService) FetchVSphereCluster(c client.Client, cluster *clust
 		Namespace: machineContext.GetObjectMeta().Namespace,
 		Name:      cluster.Spec.InfrastructureRef.Name,
 	}
-	err := c.Get(context.Background(), key, vsphereCluster)
+	err := v.Client.Get(ctx, key, vsphereCluster)
 
-	ctx.VSphereCluster = vsphereCluster
-	return ctx, err
+	machineCtx.VSphereCluster = vsphereCluster
+	return machineCtx, err
 }
 
 // ReconcileDelete reconciles delete events for VM Operator VM.
-func (v *VmopMachineService) ReconcileDelete(machineCtx capvcontext.MachineContext) error {
+func (v *VmopMachineService) ReconcileDelete(ctx context.Context, machineCtx capvcontext.MachineContext) error {
+	log := ctrl.LoggerFrom(ctx)
 	supervisorMachineCtx, ok := machineCtx.(*vmware.SupervisorMachineContext)
 	if !ok {
 		return errors.New("received unexpected SupervisorMachineContext type")
 	}
-	supervisorMachineCtx.Logger.V(2).Info("Destroying VM")
+	log.V(2).Info("Destroying VM")
 
 	// If debug logging is enabled, report the number of vms in the cluster before and after the reconcile
-	if supervisorMachineCtx.Logger.V(5).Enabled() {
-		vms, err := getVirtualMachinesInCluster(supervisorMachineCtx)
-		supervisorMachineCtx.Logger.Info("Trace Destroy PRE: VirtualMachines", "vmcount", len(vms), "error", err)
+	if log.V(5).Enabled() {
```

Comment:
I'm fine with keeping this as-is for now - but let's add a TODO to change the way this logging works - we shouldn't be using `if log.V(5).Enabled()` IMO, and we should avoid deferred logging if possible